nsv13/code/modules/squads/squad_vendor.dm

```diff
@@ -63,71 +71,110 @@
 /obj/machinery/squad_vendor/ui_act(action, params)
 	if(..())
 		return
-	var/datum/squad_loadout/S = locate(params["item_id"])
+	var/datum/squad_loadout/loadout = locate(params["item_id"])
 	var/mob/living/carbon/human/H = usr
 	if(!istype(H) || !H.squad)
 		return FALSE
 	switch(action)
 		if("vend")
-			if(!S)
+			if(!loadout)
 				return FALSE
 			if(!H.squad)
 				return FALSE
 			var/obj/item/storage/box/squad_kit/box = new(get_turf(src))
-			var/loaned_amount = 0 //How much gear have we loaned them?
-			for(var/path in S.items)
+			var/list/must_return = loadout.items.Copy()
+			//If they don't have weapons clearance, don't expect them to return guns that they don't get vended.
+			if(!H.squad.weapons_clearance)
+				for(var/banned in requires_weapons_clearance)
+					for(var/path in must_return)
+						if(ispath(path, banned))
+							must_return -= path
+			for(var/path in must_return)
```

Comment:
Can you make sure to exclude consumable items from this as well? It seems unfair to make them pay thousands of credits for using the consumables we gave them

Reply:
It doesn't do any kind of recursive check to make them return consumables - all they have to do is return the box they came in